scripts/util_functions.sh

```diff
@@ -553,11 +553,13 @@ find_magisk_apk() {
   [ -z $APK ] && APK=/data/adb/magisk.apk
   [ -f $APK ] || APK=/data/magisk/magisk.apk
   [ -f $APK ] || APK=/data/app/com.topjohnwu.magisk*/*.apk
+  [ -f $APK ] || APK=/data/app/*/com.topjohnwu.magisk*/*.apk
   if [ ! -f $APK ]; then
     DBAPK=$(magisk --sqlite "SELECT value FROM strings WHERE key='requester'" 2>/dev/null | cut -d= -f2)
-    [ -z $DBAPK ] && DBAPK=$(strings /data/adb/magisk.db | grep -E '^.requester.' | cut -c11-)
+    [ -z $DBAPK ] && DBAPK=$(strings /data/adb/magisk.db | grep -E 'requester.' | cut -dr -f3)
```

Comment:
Even better, `grep -oE 'requester.*' | cut -c10-` will handle all of these cases. 👍

Reply:
Oh wait, there's that extra line in the db which is only "requester" now. My original one covered it since the other one had the preceding char, but these ones all do not, so there's an extra \n in the output.

`grep -oE 'requester.*' | sed -e 's/requester//' -e '/^$/d'` would fix it all, or if you can come up with some other clean way then all good. 😛

Reply:
Here's a good one to ensure we match non-whitespace, then retain the rest of the line with grep: `grep -oE 'requester.\S*.*' | cut -d10-` 🙃

Reply:
The last one should be `grep -oE 'requester.\S*.*' | cut -c10-`, but it would include whitespace following the word requester.
I don't know if it can ever happen, but to avoid this we need something like:
`grep 'requester.' | sed 's/.*requester[[:blank:]]*//g'`
The dot after `requester` makes sure it's followed by a character not a newline and the `sed`command removes everything before `requester` and all whitespace following it.
And it does not rely on an exact position to cut the rest of the line.
On my test-device I have a preceding character again (after unhiding and repacking Magisk-app) and even two `r`'s in the app-name and it works as expected.

Reply:
Yes sorry, typo, I meant -c10-. I don't think there will ever be whitespace following the word requester; it's always gone straight into the app name, so mine should work fine, but whatever works without overengineering it will be fine. 🙂

Reply:
Alright, simplest should be this then, I think?:

`grep -oE 'requester..*' | cut -c10-`

Reply:
Yes, but I think the `-E` is not really needed, because we do not use extended regex here, but it does not harm, so it can be kept to be on the safe side.

Reply:
Wanna force push that and then I can resolve this and approve? Thanks!